.github/workflows/codecov.yml

```diff
@@ -0,0 +1,32 @@
+name: test and coverage
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Run tests and collect coverage
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+
+      - name: Install dependencies
+        run: go mod download
+
+      - name: Run tests
+        run: go test -race -coverprofile=coverage.txt -covermode=atomic `go list ./... | grep -v examples`
```

Comment:
```suggestion
        run: go test -race -coverprofile=coverage.txt -covermode=atomic `go list ./... `
```

Reply:
@arttet This change makes the examples directory not be included in the test coverage. Can you explain why you made this change? I was trying to make sure that the examples directory doesn't mess up the overall test coverage numbers."